classes/forum.class.php

```diff
@@ -129,4 +129,48 @@ public function show_other_posts_links($courseid, $userid) {
 
         return false;
     }
+
+    /**
+     * Create the submission event data needed to queue a submission if Turnitin is enabled after a submission.
+     *
+     * @param array $linkarray Data passed by Moodle that belongs to a submission.
+     * @return array Event data.
+     * @throws coding_exception
+     * @throws dml_exception
+     */
+    public function create_submission_event_data($linkarray) {
+        global $USER;
+
+        $cm = get_coursemodule_from_id('', $linkarray['cmid']);
+
+        $eventdata = array();
+        $eventdata['contextinstanceid'] = $linkarray['cmid'];
+
+        $eventdata['eventtype'] = 'assessable_submitted';
```

Comment:
Is this event type correct? Is that not only for assign module type?

Reply:
I thought it was the same for all modules but I'll check.

Reply:
I can confirm it is the same event type for all three modules. I double checked what is passed through for a submission and it looks like I didn't need to as we define this event type for each one in the observer class anyway.